pkg/ccl/backupccl/restore_planning.go

```diff
@@ -392,12 +392,28 @@ func allocateDescriptorRewrites(
 					}
 					parentID = newParentID
 				}
-				// Check that the table name is _not_ in use.
+
+				// If we are restoring the table into an existing schema in the target
+				// database, we must ensure that the table name is _not_ in use.
 				// This would fail the CPut later anyway, but this yields a prettier error.
-				tableName := tree.NewUnqualifiedTableName(tree.Name(table.GetName()))
-				err := descs.CheckObjectNameCollision(ctx, col, txn.KV(), parentID, table.GetParentSchemaID(), tableName)
-				if err != nil {
-					return err
+				//
+				// If we are restoring the table into a schema that is also being
+				// restored then we do not need to check for collisions as the backing
+				// up cluster would enforce uniqueness of table names in a schema.
+				rw, ok := descriptorRewrites[table.GetParentSchemaID()]
```

Comment:
nit: rename `ok` to `restoringParentSchema`?

Reply:
also, I think i'm confused: it seems we're conducting this collision check only if we're also restoring its schema which is in `descriptorRewrites`. That seems to contradict the comment above. We should only run the check if the table's schema's ID is _not_ in `descriptorRewrites`, correct?

Reply:
Both schemas that are being restored and existing schemas in the restoring cluster are in `descriptorRewrites`. The presence of the schema in `descriptorRewrites` is not what determines if we are going to create a new descriptor or map to an existing one. It is the `ToExisting` field on the rewrite that is used to determine this. Does that make sense?